File: app/build 4/tmp/kapt3/stubs/debug/com/okankkl/movieapp/ui/home_screen/HomeFragment.java
package com.okankkl.movieapp.ui.home_screen;

@dagger.hilt.android.AndroidEntryPoint
@kotlin.Metadata(mv = {1, 9, 0}, k = 1, xi = 48, d1 = {"\u0000f\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0004\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0010\u0002\n\u0000\n\u0002\u0010 \n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\b\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0006\b\u0007\u0018\u00002\u00020\u0001B\u0005\u00a2\u0006\u0002\u0010\u0002J&\u0010\u0010\u001a\u00020\u00112\f\u0010\u0012\u001a\b\u0012\u0004\u0012\u00020\u00140\u00132\u0006\u0010\u0015\u001a\u00020\u00162\u0006\u0010\u0017\u001a\u00020\u0018H\u0002J\u0018\u0010\u0019\u001a\u00020\u00112\u0006\u0010\u001a\u001a\u00020\u001b2\u0006\u0010\u001c\u001a\u00020\u001dH\u0002J&\u0010\u001e\u001a\u0004\u0018\u00010\u001f2\u0006\u0010 \u001a\u00020!2\b\u0010\"\u001a\u0004\u0018\u00010#2\b\u0010$\u001a\u0004\u0018\u00010%H\u0016J\b\u0010&\u001a\u00020\u0011H\u0016J\b\u0010\'\u001a\u00020\u0011H\u0016J\b\u0010(\u001a\u00020\u0011H\u0016J\u001a\u0010)\u001a\u00020\u00112\u0006\u0010*\u001a\u00020\u001f2\b\u0010$\u001a\u0004\u0018\u00010%H\u0016R\u0010\u0010\u0003\u001a\u0004\u0018\u00010\u0004X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0016\u0010\u0005\u001a\u0004\u0018\u00010\u00048BX\u0082\u0004\u00a2\u0006\u0006\u001a\u0004\b\u0006\u0010\u0007R\u000e\u0010\b\u001a\u00020\tX\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u001b\u0010\n\u001a\u00020\u000b8BX\u0082\u0084\u0002\u00a2\u0006\f\n\u0004\b\u000e\u0010\u000f\u001a\u0004\b\f\u0010\r\u00a8\u0006+"}, d2 = {"Lcom/okankkl/movieapp/ui/home_screen/HomeFragment;", "Landroidx/fragment/app/Fragment;", "()V", "_binding", "Lcom/okankkl/movieapp/databinding/FragmentHomeBinding;", "binding", "getBinding", "()Lcom/okankkl/movieapp/databinding/FragmentHomeBinding;", "scope", "Lkotlinx/coroutines/CoroutineScope;", "viewModel", "Lcom/okankkl/movieapp/ui/home_screen/HomeFragmentViewModel;", "getViewModel", "()Lcom/okankkl/movieapp/ui/home_screen/HomeFragmentViewModel;", "viewModel$delegate", "Lkotlin/Lazy;", "fillAdapter", "", "movieList", "", "Lcom/okankkl/movieapp/domain/model/Movie;", "adapter", "Lcom/okankkl/movieapp/ui/home_screen/MovieListAdapter;", "movieListType", "Lcom/okankkl/movieapp/util/MovieListType;", "navigateMovieDetail", "navController", "Landroidx/navigation/NavController;", "movieId", "", "onCreateView", "Landroid/view/View;", "inflater", "Landroid/view/LayoutInflater;", "container", "Landroid/view/ViewGroup;", "savedInstanceState", "Landroid/os/Bundle;", "onDestroy", "onPause", "onResume", "onViewCreated", "view", "app_debug"})
public final class HomeFragment extends androidx.fragment.app.Fragment {
    @org.jetbrains.annotations.Nullable
    private com.okankkl.movieapp.databinding.FragmentHomeBinding _binding;
    @org.jetbrains.annotations.NotNull
    private final kotlinx.coroutines.CoroutineScope scope = null;
    @org.jetbrains.annotations.NotNull
    private final kotlin.Lazy viewModel$delegate = null;
    
    public HomeFragment() {
        super();
    }
    
    private final com.okankkl.movieapp.databinding.FragmentHomeBinding getBinding() {
        return null;
    }
    
    private final com.okankkl.movieapp.ui.home_screen.HomeFragmentViewModel getViewModel() {
        return null;
    }
    
    @java.lang.Override
    @org.jetbrains.annotations.Nullable
    public android.view.View onCreateView(@org.jetbrains.annotations.NotNull
    android.view.LayoutInflater inflater, @org.jetbrains.annotations.Nullable
    android.view.ViewGroup container, @org.jetbrains.annotations.Nullable
    android.os.Bundle savedInstanceState) {
        return null;
    }
    
    @java.lang.Override
    public void onViewCreated(@org.jetbrains.annotations.NotNull
    android.view.View view, @org.jetbrains.annotations.Nullable
    android.os.Bundle savedInstanceState) {
    }
    
    @java.lang.Override
    public void onResume() {
    }
    
    @java.lang.Override
    public void onPause() {
    }
    
    @java.lang.Override
    public void onDestroy() {
    }
    
    private final void fillAdapter(java.util.List<com.okankkl.movieapp.domain.model.Movie> movieList, com.okankkl.movieapp.ui.home_screen.MovieListAdapter adapter, com.okankkl.movieapp.util.MovieListType movieListType) {
    }
    
    private final void navigateMovieDetail(androidx.navigation.NavController navController, int movieId) {
    }
}

File: app/src/main/java/com/okankkl/movieapp/ui/home_screen/HomeFragment.kt
package com.okankkl.movieapp.ui.home_screen

import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.viewModels
import androidx.recyclerview.widget.LinearLayoutManager
import com.okankkl.movieapp.databinding.FragmentHomeBinding
import com.okankkl.movieapp.util.MovieListType
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.collect
import kotlinx.coroutines.flow.collectLatest
import kotlinx.coroutines.flow.firstOrNull
import kotlinx.coroutines.launch
import java.time.LocalDateTime
import java.time.temporal.ChronoUnit

@AndroidEntryPoint
class HomeFragment : Fragment()
{
    private var _binding: FragmentHomeBinding? = null
    private val binding get() = _binding
    private val scope = CoroutineScope(Dispatchers.IO)
    val viewModel: HomeFragmentViewModel by viewModels()
    
    override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View?
    {
        _binding = FragmentHomeBinding.inflate(inflater)
        val view = binding?.root
        
        return view
    }
    
    override fun onViewCreated(view: View, savedInstanceState: Bundle?)
    {
        super.onViewCreated(view, savedInstanceState)
        
        val popularMoviesAdapter = MovieListAdapter{ movieId ->
            Log.d("MOVIES","movide id = $movieId")
        }
        val nowPlayingMoviesAdapter = MovieListAdapter{ movieId ->
            Log.d("MOVIES","movide id = $movieId")
        }
        val upComingMoviesAdapter = MovieListAdapter{ movieId ->
            Log.d("MOVIES","movide id = $movieId")
        }
        val topRatedMoviesAdapter = MovieListAdapter{ movieId ->
            Log.d("MOVIES","movide id = $movieId")
        }
        
        binding?.apply {
            popularMoviesRecyclerView.layoutManager = LinearLayoutManager(view.context,LinearLayoutManager.HORIZONTAL,false)
            nowPlayingMoviesRecyclerView.layoutManager = LinearLayoutManager(view.context,LinearLayoutManager.HORIZONTAL,false)
            upComingMoviesRecyclerView.layoutManager =  LinearLayoutManager(view.context,LinearLayoutManager.HORIZONTAL,false)
            topRatedMoviesRecyclerView.layoutManager =  LinearLayoutManager(view.context,LinearLayoutManager.HORIZONTAL,false)
            
            popularMoviesRecyclerView.adapter = popularMoviesAdapter
            nowPlayingMoviesRecyclerView.adapter = nowPlayingMoviesAdapter
            upComingMoviesRecyclerView.adapter = upComingMoviesAdapter
            topRatedMoviesRecyclerView.adapter = topRatedMoviesAdapter
            
            viewAllPopularMoviesTxt.setOnClickListener {
            
            }
            viewAllNowPlayingMoviesTxt.setOnClickListener {
            
            }
            viewAllUpComingMoviesTxt.setOnClickListener {
            }
            viewAllTopRatedMoviesTxt.setOnClickListener {
            
            }
        }
        
        
        scope.launch(Dispatchers.Main) {
            viewModel.state
            viewModel.state.collect { state ->
                val popularMovies = state.contentList.filter { it.movieListType == MovieListType.Popular }
                val nowPlayingMovies = state.contentList.filter { it.movieListType == MovieListType.NowPlaying }
                val upComingMovies = state.contentList.filter { it.movieListType == MovieListType.Upcoming }
                val topRatedMovies = state.contentList.filter { it.movieListType == MovieListType.TopRated }
                
                popularMoviesAdapter.setMovieList(popularMovies)
                nowPlayingMoviesAdapter.setMovieList(nowPlayingMovies)
                upComingMoviesAdapter.setMovieList(upComingMovies)
                topRatedMoviesAdapter.setMovieList(topRatedMovies)
                
                popularMoviesAdapter.notifyDataSetChanged()
                nowPlayingMoviesAdapter.notifyDataSetChanged()
                upComingMoviesAdapter.notifyDataSetChanged()
                topRatedMoviesAdapter.notifyDataSetChanged()
            }
        }
        
    }
    
    override fun onResume()
    {
        super.onResume()
        val viewModel: HomeFragmentViewModel by viewModels()
        viewModel.getMovies()
    }
    
    override fun onPause()
    {
        super.onPause()
        scope.launch(Dispatchers.IO) {
            viewModel.clearMoviesFromRoom()
            viewModel.addMoviesToRoom()
        }
    }
    
    override fun onDestroy()
    {
        super.onDestroy()
        _binding = null
    }
    
}